codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustReservedWords.kt

```diff
@@ -55,7 +56,20 @@ class RustReservedWordSymbolProvider(private val base: RustSymbolProvider, priva
                 "Self" -> "SelfValue"
                 // Real models won't end in `_` so it's safe to stop here
                 "SelfValue" -> "SelfValue_"
-                else -> baseName
+                else -> reservedWordReplacedName
+            }
+
+            container is EnumShape || container.hasTrait<EnumTrait>() -> when (baseName) {
+                // Self cannot be used as a raw identifier, so we can't use the normal escaping strategy
+                // https://internals.rust-lang.org/t/raw-identifiers-dont-work-for-all-identifiers/9094/4
+                "Self" -> "SelfValue"
+                // Real models won't end in `_` so it's safe to stop here
+                "SelfValue" -> "SelfValue_"
+                // Unknown is used as the name of the variant containing unexpected values
+                "Unknown" -> "UnknownValue"
```

Comment:
This is somewhat a spooky action at a distance - perhaps worth putting `Unknown` in a constant and refer to it here and where we generated the unexpected variant?

Reply:
I think this comment applies more to #2382, where this is more heavily refactored. I'll add the comment there.